components/searchBar/index.js

```diff
@@ -59,6 +59,53 @@ export default class searchBar extends Component {
     }
   }
 
+  async getNearbyPlaces(value) {
+    const markers = [];
+    if (value.toLowerCase().includes('near sgw') || value.toLowerCase().includes('near loy')) {
+      const formattedVal = value.substring(0, value.indexOf('near')).trim().replace(' ', '+');
+      const key = 'AIzaSyCqNODizSqMIWbKbO8Iq3VWdBcK846n_3w';
+      let url;
+      if (value.toLowerCase().includes('sgw')) {
+        this.setState({
+          region: {
+            latitude: 45.492409,
+            longitude: -73.582153,
+            latitudeDelta: 0.05,
+            longitudeDelta: 0.05
+          }
+        });
+        url = `https://maps.googleapis.com/maps/api/place/textsearch/json?query=${formattedVal}&location=45.492409,-73.582153&radius=2&key=${key}`;
+      } else if (value.toLowerCase().includes('loy')) {
```

Comment:
an else would be better to indicate that it must either be SGW or LOY in this function. On top of that, move the url out of the inner ifs and use the state variables to set the location.

Reply:
A strange bug happens when I remove the url and use the states to fill in the longitude and latitude

Reply:
@Belize22 thank you Peter for solving that race condition.